### PR TITLE
hmac validation and upgrade of pytest versions

### PR DESCRIPTION
**Pull request by Grajo**
HMAC validation added to the Client functionality.
Also, fix added as Tests were failing to run due to old version mismatch current python version.

### CODE REVIEW

File: pycronofy/client.py [hunk 1]
```diff
@@ -897,3 +899,20 @@ def delete_availability_rule(self, availability_rule_id):
         :param string availability_rule_id: ID of the Availability Rule to delete.
         """
         self.request_handler.delete(endpoint='availability_rules/%s' % availability_rule_id)
+
+    def hmac_match(self, hmac_string, body):
+        if hmac_string is None or not hmac_string:
```
Comment (Nevett) [inline pycronofy/client.py:906]: Please write some doc comments similar to the other methods

Comment (Grajo) [reply pycronofy/client.py]: Suggestion applied! :)

**Comment by Grajo** [comment]
Hey @Nevett after rebasing this project with master (which contained latest changes from GitHub-actions-migration) I realised that the CI tests where failing, after some investigation I figure out that it was due to a problem with the python versions 2.7 was deprecated and 3.6 didn't have the Mac.digest functionality, as this last one was available from 3.7. So I have modified the GitHub Actions python versions to [3.7,3.8,3.9] and seems like is working fine now! Let me know if you have any concerns about it, please?

**Comment by Nevett** [comment]
> Hey @Nevett after rebasing this project with master (which contained latest changes from GitHub-actions-migration) I realised that the CI tests where failing, after some investigation I figure out that it was due to a problem with the python versions 2.7 was deprecated and 3.6 didn't have the Mac.digest functionality, as this last one was available from 3.7. So I have modified the GitHub Actions python versions to [3.7,3.8,3.9] and seems like is working fine now! Let me know if you have any concerns about it, please?

Let's try to continue supporting 2.7 as discussed, and take a look at `compare_digest` instead of quality comparison to avoid timing attacks.

File: pycronofy/client.py [hunk 2]
```diff
@@ -939,3 +943,24 @@ def delete_availability_rule(self, availability_rule_id):
...
+
+        :hmac_string: A String containing comma-separated values describing HMACs of the notification taken from the
+                        Cronofy-HMAC-SHA256 header.
+        :body: A String of the body of the notification.
+        :return: TRUE if one of the HMAC provided matches the one calculated using the client secret, otherwise FALSE.
+        """
+        if hmac_string is None or not hmac_string:
+            return False
+
+        # In order to support oython 2.7 we generate the hmac and calculate the digest in two steps using `new` and `digest`
```
Comment (Nevett) [inline pycronofy/client.py:958]:
```suggestion
        # In order to support Python 2.7 we generate the hmac and calculate the digest in two steps using `new` and `digest`
```

**Pull request merged**